src/ru/geekbrains/java3/Box.java

```diff
@@ -0,0 +1,68 @@
+package ru.geekbrains.java3;
+
+import java.util.ArrayList;
+
+public class Box<E extends Fruit> {
+    E currentFruit;
+    private String boxName;
+    private ArrayList<E> boxContent = new ArrayList<>();
+    private String fruitType;
+
+    public Box(String name) {
+        this.boxName = name;
+    }
+
+    public void addFruit(E fruit) {
+        if(boxContent.isEmpty()) {
+            boxContent.add(fruit);
+            currentFruit = fruit;
+            System.out.printf("В коробке %s %d объектов %s\n", boxName, boxContent.size(), currentFruit.getType());
+        }
+        else {
+            if (fruit.getType() == this.currentFruit.getType()) {
+                boxContent.add(fruit);
+
+                System.out.printf("В коробке %s %d объектов %s\n", boxName, boxContent.size(), currentFruit.getType());
+            } else System.out.printf("Вы не можете добавить объект %s в корзину с объектами %s\n", fruit.getType(), currentFruit.getType());
+        }
+    }
+
+    public float getWeight(){
+        return boxContent.size() * currentFruit.getWeight();
+    }
+
+    public Integer getBoxSize() {
+        return boxContent.size();
+    }
+
+    public ArrayList<E> getBoxFruits() {
+        return boxContent;
+    }
+
+    public boolean compare(Box comparedBox) {
```

Comment:
```suggestion
    public boolean compare(Box<?> comparedBox) {
```
если класс типизированный, то лучше везде указывать для него дженерик, чтобы явно указать, какой тип мы можем здесь использовать